File: templates/ApplicationCommand/ApplicationCommand.py
import sublime_plugin

from ...pylib.JSApplicationCommand import JSApplicationCommand

class {{commandName}}Command(JSApplicationCommand, sublime_plugin.ApplicationCommand):

  {{#run}}
  def run(self, **args):
    super({{commandName}}Command, self).run(**args)
  {{/run}}
  {{^run}}
  def run(self, **args):
    pass
  {{/run}}

  {{#is_enabled}}
  def is_enabled(self, **args):
    super({{commandName}}Command, self).is_enabled(**args)
  {{/is_enabled}}
  {{^is_enabled}}
  def is_enabled(self, **args):
    return True
  {{/is_enabled}}

  {{#is_visible}}
  def is_visible(self, **args):
    super({{commandName}}Command, self).is_visible(**args)
  {{/is_visible}}
  {{^is_visible}}
  def is_visible(self, **args):
    return True
  {{/is_visible}}

  {{#is_checked}}
  def is_checked(self, **args):
    super({{commandName}}Command, self).is_checked(**args)
  {{/is_checked}}
  {{^is_checked}}
  def is_checked(self, **args):
    return True
  {{/is_checked}}
  
  {{#description}}
  def description(self, **args):
    super({{commandName}}Command, self).description(**args)
  {{/description}}
  {{^description}}
  def description(self, **args):
    return None
  {{/description}}

File: jslib/ApplicationCommand.js
// @flow

const applicationCommands = require('./applicationCommandList.js'),
      SublimeObject = require('./SublimeObject.js')

/**
 * [sublime_plugin.ApplicationCommand Class](https://www.sublimetext.com/docs/3/api_reference.html#sublime_plugin.ApplicationCommand).
 */
class ApplicationCommand extends SublimeObject {

  constructor () {
    super(null, null, true)
    applicationCommands[this.constructor.name] = this
  }

  /**
   * Called when the command is run.
   */
  async run (args /*: Object*/, step /*: StepObject*/) /*: Promise<void>*/ {

  }

  /**
   * Returns ```true``` if the command is able to be run at this time. The default implementation simply always returns ```true```.
   */
  async is_enabled (args /*: Object*/, step /*: StepObject*/) /*: Promise<boolean>*/ {
    return true
  }

  /**
   * Returns ```true``` if the command should be shown in the menu at this time. The default implementation always returns ```true```.
   */
  async is_visible (args /*: Object*/, step /*: StepObject*/) /*: Promise<boolean>*/ {
    return true
  }

  /**
   * Returns ```true``` if a checkbox should be shown next to the menu item. The ```.sublime-menu``` file must have the checkbox attribute set to ```true``` for this to be used.
   */
  async is_checked (args /*: Object*/, step /*: StepObject*/) /*: Promise<boolean>*/ {
    return true
  }

  /**
   * Returns a description of the command with the given arguments. Used in the menu, if no caption is provided. Return ```null``` to get the default description.
   */
  async description (args /*: Object*/, step /*: StepObject*/) /*: Promise<string | null>*/ {
    return null
  }

}

module.exports = ApplicationCommand